src/Datadog.Trace/Sampling/RegexSamplingRule.cs

```diff
@@ -8,11 +8,14 @@ namespace Datadog.Trace.Sampling
     internal class RegexSamplingRule : ISamplingRule
     {
         private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.For<RegexSamplingRule>();
+        private static readonly TimeSpan RegexTimeout = TimeSpan.FromMilliseconds(500);
```

Comment:
Under load, I can imagine this being too short a timeout. I know 500 ms is a lot of time, but I'd rather see it at at least 1 second.

Reply:
Why not 5?

Reply:
I'd be happy with at least 1 second, maybe 2 or 3 seconds. :) Splittling hairs, haha